File: src/RepositoryModules.CS/SecondaryCondition.cs
using System;

namespace RepositoryModules
{
    sealed class SecondaryCondition
    {
        public string PrimaryLabelItemNumber { get; }

        public string SecondaryItemNumber { get; }

        public SecondaryCondition(string primaryLabelItemNumber, string secondaryItemNumber)
        {
            if (string.IsNullOrWhiteSpace(primaryLabelItemNumber)) throw new ArgumentException($"'{nameof(primaryLabelItemNumber)}' を null または空白にすることはできません。", nameof(primaryLabelItemNumber));
            if (string.IsNullOrWhiteSpace(secondaryItemNumber)) throw new ArgumentException($"'{nameof(secondaryItemNumber)}' を null または空白にすることはできません。", nameof(secondaryItemNumber));
            
            PrimaryLabelItemNumber = primaryLabelItemNumber;
            SecondaryItemNumber = secondaryItemNumber;
        }
    }
}

File: src/RepositoryModules.VB/SecondaryCondition.vb
Friend Class SecondaryCondition
    Public ReadOnly Property PrimaryLabelItemNumber As String

    Public ReadOnly Property SecondaryItemNumber As String

    Public Sub New(PrimaryLabelItemNumber As String, SecondaryItemNumber As String)

        If String.IsNullOrWhiteSpace(PrimaryLabelItemNumber) Then

            Throw New ArgumentException($"'{NameOf(PrimaryLabelItemNumber)}' を null または空白にすることはできません。", NameOf(PrimaryLabelItemNumber))
        End If

        If String.IsNullOrWhiteSpace(SecondaryItemNumber) Then

            Throw New ArgumentException($"'{NameOf(SecondaryItemNumber)}' を null または空白にすることはできません。", NameOf(SecondaryItemNumber))
        End If

        Me.PrimaryLabelItemNumber = PrimaryLabelItemNumber
        Me.SecondaryItemNumber = SecondaryItemNumber
    End Sub
End Class
